## Ejercicio 3: Análisis exploratorio y transformación de datos

En este ejercicio se realiza un análisis exploratorio de un conjunto de datos mediante estadísticas descriptivas y el cálculo de correlaciones.

Además, se aplica una transformación de variables categóricas a formato numérico para facilitar su uso en análisis y modelos de minería de datos.

In [1]:
import pandas as pd

df3 = pd.DataFrame({
    "ID": [1, 2, 3, 4],
    "Nombre": ["Juan", "María", "Pedro", "Ana"],
    "Edad": [32, 28, 40, 35],
    "Categoria_Cliente": ["Oro", "Plata", "Platino", "Oro"],
    "Ingresos": [45000, 55000, 70000, 60000]
})

print("Dataset del ejercicio 3:\n")
print(df3.to_string(index=False))

Dataset del ejercicio 3:

 ID Nombre  Edad Categoria_Cliente  Ingresos
  1   Juan    32               Oro     45000
  2  María    28             Plata     55000
  3  Pedro    40           Platino     70000
  4    Ana    35               Oro     60000


In [4]:
print("Estadísticas descriptivas:\n")
print(df3.describe().round(2))

Estadísticas descriptivas:

         ID   Edad  Ingresos
count  4.00   4.00      4.00
mean   2.50  33.75  57500.00
std    1.29   5.06  10408.33
min    1.00  28.00  45000.00
25%    1.75  31.00  52500.00
50%    2.50  33.50  57500.00
75%    3.25  36.25  62500.00
max    4.00  40.00  70000.00


### Interpretación de las estadísticas descriptivas

- La edad promedio de los clientes es de **33.75 años**.
- El ingreso promedio es de **57,500**.
- La edad mínima es **28** y la máxima es **40**.
- El ingreso mínimo es **45,000** y el máximo es **70,000**.
- La mediana de edad es **33.50** y la mediana de ingresos es **57,500**.
- No se observan valores faltantes en las variables numéricas analizadas.

In [5]:
from sklearn.preprocessing import OrdinalEncoder

orden_categoria = [["Plata", "Oro", "Platino"]]
encoder = OrdinalEncoder(categories=orden_categoria)

df3["Categoria_Cod"] = encoder.fit_transform(df3[["Categoria_Cliente"]])

print("Resultado de la codificación de 'Categoria_Cliente':\n")
print(df3[["Nombre", "Categoria_Cliente", "Categoria_Cod"]].to_string(index=False))

Resultado de la codificación de 'Categoria_Cliente':

Nombre Categoria_Cliente  Categoria_Cod
  Juan               Oro            1.0
 María             Plata            0.0
 Pedro           Platino            2.0
   Ana               Oro            1.0


In [6]:
correlacion = df3[["Edad", "Ingresos"]].corr().round(2)

print("Correlación entre Edad e Ingresos:\n")
print(correlacion)

Correlación entre Edad e Ingresos:

          Edad  Ingresos
Edad      1.00      0.74
Ingresos  0.74      1.00


### Interpretación de la correlación

Se observa una correlación positiva de **0.74** entre la edad y los ingresos.

Esto indica que, en este conjunto de datos, a mayor edad, tienden a aumentar los ingresos. Sin embargo, la relación no es perfecta, por lo que pueden existir otros factores que influyan en los ingresos.

### Conclusión

- La correlación obtenida sugiere una relación moderada-alta entre edad e ingresos, lo que podría ser relevante para modelos predictivos en futuros análisis.